### ISSUE TITLE
Ticket commitments (and other null data scripts) should not display an address

### ISSUE THREAD

**Issue by jrick** [decred/dcrdata#490]
Addresses are encodings of specific forms of scripts.  Every address can be converted to a script, but not all scripts have an associated address.

One form of a script without an address is the null data script, also called an OP_RETURN script.  These are unspendable outputs which only carry data.

Currently, dcrdata displays an address for null data scripts (so similar outputs can be queried?).  This is highly misleading, and we especially don't want people creating new txs paying to these addresses (which won't result in the same script anyways).

One such example pulled from a recent block is https://explorer.dcrdata.org/tx/eb568754b57323fd16e027301e7f1030961999b6f23bc9364dcc3d2c9af726d0.  The second output (index 1) is a ticket purchase commitment with the address Dsnja54PAoyYTmKAtDmTZ3S2Mg4dGwPmfX1.  I'm not sure how this address is created, but I do know it's wrong since there isn't any address for this script.

For reference: https://github.com/decred/dcrd/blob/75432b3c9621687f2fb71b5cb2875e7aa13f24c1/txscript/standard.go#L1270-L1272

**Comment by davecgh** [comment]
This is only partially true.  In general, `OP_RETURN` scripts should not have an address associated with them, since, as you note, they are provably unspendable.

However, in the case of ticket commitments, the consensus rules enforce the data there to contain a commitment to either a p2pkh or a p2sh address.  Hence, I believe what dcrdata is doing is displaying the address that is embedded within the commitment.

**Comment by jrick** [comment]
Yes, the data push does include the p2pkh or p2sh address being committed to, and I suppose that would make some sense to display it in that case.

Unsure if I feel if it is a good idea to display that address or not.
